engines/bops_api/app/services/bops_api/application/submission_redaction_service.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module BopsApi
+  module Application
+    class SubmissionRedactionService
+      class RedactionError < StandardError; end
+
+      REDACTION_TEXT = "REDACTED"
+
+      FIELDS_TO_REDACT = [
+        ["data", "applicant", "phone", "primary"],
+        ["data", "applicant", "phone", "secondary"],
+        ["data", "applicant", "email"],
+        ["data", "applicant", "agent", "phone", "primary"],
+        ["data", "applicant", "agent", "phone", "secondary"],
+        ["data", "applicant", "agent", "email"]
+      ]
+
+      def initialize(planning_application:)
+        @planning_application = planning_application
+      end
+
+      def call
+        return unless (submission = @planning_application.params_v2)
+
+        redact_fields(submission)
+      rescue => e
+        raise RedactionError, e.message
+      end
+
+      private
+
+      def redact_fields(data)
+        FIELDS_TO_REDACT.each do |field_path|
+          field = field_path.reduce(data) { |d, key| d[key] if d }
+          field&.replace(REDACTION_TEXT)
```

Comment:
Would using `dig` here be clearer? Something like `data.dig(*field_path).replace(...)`.

Reply:
Will give that a go

Reply:
Works well!